src/System.Net.Security/src/System/Net/_SSPIWrapper.cs

```diff
@@ -0,0 +1,873 @@
+
+                        default: throw NotImplemented.ByDesignWithMessage(SR.net_MethodNotImplementedException);
+                    }
+
+                    // Marshalling back returned sizes / data.
+                    for (int i = 0; i < input.Length; i++)
+                    {
+                        SecurityBuffer iBuffer = input[i];
+                        iBuffer.size = unmanagedBuffer[i].count;
+                        iBuffer.type = unmanagedBuffer[i].type;
+
+                        if (iBuffer.size == 0)
+                        {
+                            iBuffer.offset = 0;
+                            iBuffer.token = null;
+                        }
+                        else
+                            checked
+                            {
+                                // Find the buffer this is inside of.  Usually they all point inside buffer 0.
+                                int j;
+                                for (j = 0; j < input.Length; j++)
+                                {
+                                    if (buffers[j] == null)
+                                    {
+                                        continue;
+                                    }
+
+                                    byte* bufferAddress = (byte*)Marshal.UnsafeAddrOfPinnedArrayElement(buffers[j], 0);
+                                    if ((byte*)unmanagedBuffer[i].token >= bufferAddress &&
+                                        (byte*)unmanagedBuffer[i].token + iBuffer.size <= bufferAddress + buffers[j].Length)
+                                    {
+                                        iBuffer.offset = (int)((byte*)unmanagedBuffer[i].token - bufferAddress);
+                                        iBuffer.token = buffers[j];
+                                        break;
+                                    }
+                                }
+
+                                if (j >= input.Length)
+                                {
+                                    GlobalLog.Assert("SSPIWrapper::EncryptDecryptHelper", "Output buffer out of range.");
+                                    iBuffer.size = 0;
+                                    iBuffer.offset = 0;
+                                    iBuffer.token = null;
+                                }
+                            }
+
+                        // Backup validate the new sizes.
+                        GlobalLog.Assert(iBuffer.offset >= 0 && iBuffer.offset <= (iBuffer.token == null ? 0 : iBuffer.token.Length), "SSPIWrapper::EncryptDecryptHelper|'offset' out of range.  [{0}]", iBuffer.offset);
+                        GlobalLog.Assert(iBuffer.size >= 0 && iBuffer.size <= (iBuffer.token == null ? 0 : iBuffer.token.Length - iBuffer.offset), "SSPIWrapper::EncryptDecryptHelper|'size' out of range.  [{0}]", iBuffer.size);
+                    }
+
+                    if (errorCode != 0)
+                        if (Logging.On)
```

Comment:
Can we combine these ifs so that there's not a braced if within an unbraced one?